astropy/table/column.py

```diff
@@ -271,11 +273,31 @@ def __reduce__(self):
 
         return reconstruct_func, reconstruct_func_args, state
 
+    def __getslice__(self, start, stop):
+        return self.__getitem__(slice(start, stop))
+
+    
     def __getitem__(self, item):
         if isinstance(item, INTEGER_TYPES):
             return self.data[item]  # Return as plain ndarray or ma.MaskedArray
+
+        elif isinstance(item, slice):
```

Comment:
Yeah, I think let's just preserve them at the table level and see whether anyone notices that the aren't preserved when slicing columns...